flowey/flowey_lib_hvlite/src/run_cargo_build.rs

```diff
@@ -324,69 +328,32 @@ impl FlowNode for Node {
         } in requests
         {
             let mut pre_build_deps = base_pre_build_deps.clone();
-            let mut injected_env = BTreeMap::new();
 
-            pre_build_deps.extend(user_pre_build_deps);
+            let (cross_deps, injected_env) = setup_cross_arch_build(ctx, &target)?;
 
-            // when cross compiling aarch64-linux from linux, explicitly set the
-            // linker being used
-            //
-            // FUTURE: this will need tweaking flowey is being run on non-x86
-            // hosts
-            if matches!(ctx.platform(), FlowPlatform::Linux)
-                && matches!(
-                    target.architecture,
-                    target_lexicon::Architecture::Aarch64(_)
-                )
-                && matches!(
-                    target.operating_system,
-                    target_lexicon::OperatingSystem::Linux
-                )
-            {
-                // We use `gcc`'s linker for cross-compiling due to:
-                //
-                // * The special baremetal options are the same. These options
-                //   don't work for the LLVM linker,
-                // * The compiler team at Microsoft has stated that `rust-lld`
-                //   is not a production option,
-                // * The only Rust `aarch64` targets that produce
-                //   position-independent static ELF binaries with no std are
-                //   `aarch64-unknown-linux-*`.
-                pre_build_deps.push(ctx.reqv(|v| {
-                    flowey_lib_common::install_apt_pkg::Request::Install {
-                        package_names: vec!["gcc-aarch64-linux-gnu".into()],
-                        done: v,
-                    }
-                }));
-
-                // HACK: ...unless this is musl, in which case, we use the
-                // openhcl linker set in the repo's `.cargo/config.toml`
-                //
-                // Not sure if I'm a fan of the current approach to begin with,
-                // since it means _any_ musl code (not just code running in
-                // VTL2) will use the openhcl-specific musl... but whatever
-                if !matches!(target.environment, target_lexicon::Environment::Musl) {
-                    injected_env.insert(
-                        format!(
-                            "CARGO_TARGET_{}_LINKER",
-                            target.to_string().replace('-', "_").to_uppercase()
-                        ),
-                        "aarch64-linux-gnu-gcc".into(),
-                    );
-                }
-            }
+            pre_build_deps.extend(cross_deps);
+            pre_build_deps.extend(user_pre_build_deps);
 
-            // FIXME?: because we set `CC_x86_64_unknown_linux_musl` in our cargo env,
-            // we end up compining _every_ musl artifact using the openhcl musl
+            // FIXME?: because we set `CC_{arch}_unknown_linux_musl` in our cargo env,
```

Comment:
yes, this is still a real FIXME. we don't want musl builds of openvmm (e.g: for Alpine linux, or whatever) to have a dependency on the OpenHCL sysroot

Reply:
True, it doesn't seem like something that needs to be addressed in this PR though.

Reply:
oh, whoops, I totally thought you flipped the FIXME to a FIXME? 😅

can you flip this to a proper FIXME here?

Reply:
Oh haha sure